General: Split wal into multiple files.

dylanaraps opened the pull request:
This PR extends #7 and splits `wal` into multiple files based on their purpose in the original script. This will make `wal` much easier to maintain in the future.

@danielx, can you take a look at this?

danielx commented:
Looks good, but i would make two changes:

* Remove the use of globals to make the functions easier to test and reason about.
* Make module version available in`__init__.py` by import or declaring it there.

dylanaraps commented:
> Remove the use of globals to make the functions easier to test and reason about.

I'm currently in the process of doing this.

> Make module version available in__init__.py by import or declaring it there.

Good idea, I'll get onto this.

Thanks for the advice!

dylanaraps commented:
I've removed all globals (minus the 3 constants which are ok imo.) and I've added `__version__` to `__init__.py`. 

How does this PR look now? @danielx

danielx commented on pywal/gen_colors.py:
```diff
@@ -0,0 +1,137 @@
+    return colors
+
+
+def sort_colors(colors):
```
Function body could be shortened to

```python
return colors[:1] + colors[9:] + colors[8:]
```

And If the length of `colors` might be greater than 16
```python
return colors[:1] + colors[9:16] + colors[8:16]
```

danielx commented:
Looks good to me 👍

dylanaraps commented:
Thanks for taking another look I apreciate it!
